Sources/YPersistence/Manager/PersistenceManager+Sync.swift

```diff
@@ -63,7 +65,100 @@ extension PersistenceManager {
         )
         return try fetchRecords(
             predicate: predicate,
-            sortDescriptors: T.sort?.descriptors
+            sortDescriptors: T.sort?.descriptors,
+            context: context
         )
     }
+
+    /// Update records that have been marked for upload since the last sync.
+    ///
+    /// May be safely called from any thread
+    /// may only be safely accessed from the current thread.
+    /// If no context is provided, a worker context will be created.
+    /// - Parameters:
+    ///   - enity: the Core Data entity to be fetched
+    ///   - uids: array of uids to update
+    ///   - context: Optional managed context to use. Default is `nil`.
+    /// - Throws: any error executing the fetch or save request
+    public func markRecordsAsUploaded<T: SyncRecord>(
+        enity: T.Type,
+        uids: [T.UidType],
+        context: NSManagedObjectContext? = nil
+    ) throws {
+        let context = context ?? workerContext
+
+        let predicate = NSPredicate(format: "\(enity.uidKey) in %@ && \(T.isUploadedKey) == false", uids)
+        
+        let records: [T] = try fetchRecords(
+            predicate: predicate,
+            context: context
+        )
+
+        for record in records {
+            record.isUploaded = true
+        }
+
+        try context.saveChangesAndWait()
+    }
+
+    /// Update records that have been marked for deletion since the last sync.
+    ///
+    /// May be safely called from any thread
+    /// may only be safely accessed from the current thread.
+    /// If no context is provided, a worker context will be created.
+    /// - Parameters:
+    ///   - enity: the Core Data entity to be fetched
```

Comment:
```suggestion
    ///   - entity: the Core Data entity to be fetched
```
Same typo. Please fix everywhere.